src/esp/gfx_batch/Renderer.cpp

```diff
@@ -522,17 +553,47 @@ bool Renderer::addFile(const Cr::Containers::StringView filename,
         return {};
       }
 
-      const auto& flatMaterial = material->as<Mn::Trade::FlatMaterialData>();
-      materials[i] = Mn::Shaders::PhongMaterialUniform{}.setAmbientColor(
-          flatMaterial.color());
-
-      /* Untextured materials get the first reserved texture (a white pixel) */
-      if (!flatMaterial.hasTexture())
-        materialTextureTransformations[i] = {0, 0, {}};
-      else
-        materialTextureTransformations[i] = {
-            flatMaterial.texture() + textureOffset, flatMaterial.textureLayer(),
-            flatMaterial.textureMatrix()};
+      /* Set up flat shading either if the material is marked as flat or if
+         we have no lights enabled */
+      if (material->types() & Mn::Trade::MaterialType::Flat ||
+          !state_->maxLightCount) {
+        const auto& flatMaterial = material->as<Mn::Trade::FlatMaterialData>();
+        materials[i] = Mn::Shaders::PhongMaterialUniform{}
+                           .setAmbientColor(flatMaterial.color())
+                           /* Diffuse is zero so lights (if enabled) have no
+                              effect on this material  */
+                           .setDiffuseColor(0x00000000_rgbaf);
+
+        /* Untextured materials get the first reserved texture (a white
+           pixel) */
+        if (!flatMaterial.hasTexture())
+          materialTextureTransformations[i] = {0, 0, {}};
+        else
+          materialTextureTransformations[i] = {
+              flatMaterial.texture() + textureOffset,
+              flatMaterial.textureLayer(), flatMaterial.textureMatrix()};
+      } else {
+        const auto& phongMaterial =
+            material->as<Mn::Trade::PhongMaterialData>();
+        materials[i] = Mn::Shaders::PhongMaterialUniform{}
+                           .setAmbientColor(phongMaterial.diffuseColor() * 0.1f)
+                           .setDiffuseColor(phongMaterial.diffuseColor())
+            /* Specular not used (and shader compiled with NoSpecular). Much
```

Comment:
Can you explain this better? Is the problem that our existing assets have too much spec on their materials? Or is the problem that our point lighting doesn't produce nice-looking highlights?

Reply:
Same as [in case of the ambient](https://github.com/facebookresearch/habitat-sim/pull/1970#discussion_r1063620089), the PBR materials in glTF don't encode any info about "classic" specular so the renderer has to guesstimate and it usually guesses wrong. From my experience the specular "white spots" were what usually people complained about when seeing the renderer output (such as [the recent Floorplanner discussion on Slack](https://cvmlp.slack.com/archives/CFN5TAUSD/p1671556128596599)).